src/material/datepicker/date-range-input.ts

```diff
@@ -392,6 +394,16 @@ export class MatDateRangeInput<D>
     this.stateChanges.next();
   }
 
+  /** Gets the value for the aria label for the start date input. */
+  _getStartDateLabel() {
```

Comment:
I wonder if it wouldn't make more sense for the start/end date labels to be configurable either on `mat-date-range-input` or on `matStartDate`/`matEndDate`, instead of through `MatDatepickerIntl`. The problem with having it be in the provider is that it affects all range inputs in the app, but they may represent different things. E.g. in a travel site the flights search might say "Departure" and "Return" while the hotel search would say "Check in" and "Check out".

Reply:
That's good point, didn't think of that. `MatDatepickerIntl` is injectable, so if there were a Flights pages and a Hotel pages in the same app. The Flight page could inject "Departure"/"Arrival" and the hotel page could inject "Check in"/"Check out". It seems like we have this case covered, but I should definitely add a unit test for that.

Reply:
I still don't get it; I think I might not be understanding how DI works. If there were two different range inputs rendered by two different components, couldn't the two inject their own values for "Start Date"/"End Date"?

Something like injecting either `HotelMatDatepickerIntl` or `AirlineMatDatepickerIntl` depending on the need.

Reply:
I don't think it would be necessary to make the labels configurable on `mat-date-range-input `or on matStartDate/matEndDate, instead of through MatDatepickerIntl.

On the one hand, it makes sense for developers to be able to able to set the descriptions for each individual form field. On the other hand, we would also need a way to specify an aria-description on the calendar pop-up for date ranges of length 1, where the start date and end date are on the same day (e.g. "Start and end date"). In general, is there a requirement for how customizable the `matStartDate`/`matEndDate` are?